js/braziliex.js

```diff
@@ -0,0 +1,425 @@
+        let market = this.market (symbol);
+        let result = await this.privatePostCancelOrder (this.extend ({
+            'order_number': id,
+            'market': market['id'],
+        }, params));
+        return result;
+    }
+
+    async fetchOpenOrders (symbol, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        let market = this.market (symbol);
+        let orders = await this.privatePostOpenOrders (this.extend ({
+            'market': market['id'],
+        }, params));
+        return this.parseOrders (orders['order_open'], market, 'open');
+    }
+
+    async fetchClosedOrders (symbol, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        let market = this.market (symbol);
+        let orders = await this.privatePostTradeHistory (this.extend ({
+            'market': market['id'],
+        }, params));
+        return this.parseOrders (orders['trade_history'], market, 'closed');
+    }
+
+    async fetchDepositAddress (currencyCode, params = {}) {
+        await this.loadMarkets ();
+        let currency = this.currency (currencyCode);
+        let response = await this.privatePostDepositAddress (this.extend ({
+            'currency': currency['id']
+        }, params));
+        let address = this.safeString (response['deposit_address'], 'address');
+        if (!address)
+            throw new ExchangeError (this.id + ' fetchDepositAddress failed: ' + this.last_http_response);
+        return {
+            'currency': currencyCode,
+            'address': address,
+            'status': 'ok',
+            'info': response,
+        };
+    }
+
+    sign (path, api = 'public', method = 'GET', params = {}, headers = undefined, body = undefined) {
+        let url = this.urls['api'] + '/' + api + '/';
+        let query = undefined;
+        if (api == 'public') {
+            query = this.urlencode (this.omit (params, this.extractParams (path)));
+            url += this.implodeParams (path, params);
+            if (query.length)
+                url += '?' + query;
+        } else {
+            this.checkRequiredCredentials ();
+            query = this.extend ({
+                'command': path,
+                'nonce': this.nonce (),
+            }, params);
+            body = this.json (query);
+            query = this.urlencode (query);
+            let signature = this.hmac (this.encode (query), this.encode (this.secret), 'sha256');
+            headers = {
+                'Content-type': 'application/json',
+                'Key': this.apiKey,
+                'Sign': signature,
+            };
+        }
+        return { 'url': url, 'method': method, 'body': body, 'headers': headers };
+    }
+
+    async request (path, api = 'public', method = 'GET', params = {}, headers = undefined, body = undefined) {
+        let response = await this.fetch2 (path, api, method, params, headers, body)
```

Comment:
this is pretty much ready, just a few minor edits, if you want to make it pass the test: one semicolon missing in this line